File: java/src/main/java/com/iifl/api/vm/MarketFeed.java
package com.iifl.api.vm;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

public class MarketFeed {

	
	@JsonProperty(value = "ClientCode")
	private String ClientCode; 
	
	@JsonProperty(value = "Count")
	private String Count;
	
	@JsonProperty(value = "MarketFeedData")
	private List<MarketFeedd> MarketFeedData;
	
	@JsonProperty(value = "ClientLoginType")
	private String ClientLoginType;
	
	@JsonProperty(value = "LastRequestTime")
	private String LastRequestTime;
	
	@JsonProperty(value = "RefreshRate")
	private String RefreshRate;
		
		
}


 class MarketFeedd{
	@JsonProperty(value = "Exch")
	private String Exch;
	@JsonProperty(value = "ExchType")
	private String ExchType;
	@JsonProperty(value = "ScripCode")
	private String ScripCode;
}

File: Go lang/IIFL/class/MarketFeed.go
package class

import (
	"encoding/json"
	"fmt"
	"net/http"
	"strings"
)

//MarketFeed :- This API is used to fetch the market feed of a particular scrip or a set of scrips
func MarketFeed(w http.ResponseWriter, r *http.Request) {

	ClientCode := r.URL.Query().Get("ClientCode")

	config := CheckConfig()

	url := config.ServiceURL + "MarketFeed"

	method := "POST"

	header := Header{AppName: config.AppName, AppVer: config.AppVer, Key: config.Key, OsName: config.OsName, RequestCode: config.RequestCodeMarketFeed, UserID: config.UserID, Password: config.Password}

	TradeInformList1 := MarketFeedDataList{Exch: "N", ExchType: "C", ScripCode: "2885"}
	TradeInformList2 := MarketFeedDataList{Exch: "N", ExchType: "C", ScripCode: "22"}

	body := MarketFeedBody{ClientCode: ClientCode, Count: "2", MarketFeedDataList: []MarketFeedDataList{TradeInformList1, TradeInformList2}, ClientLoginType: "0", LastRequestTime: "/Date(1600248018615)/", RefreshRate: "H"}

	MarketFeedRequest := MarketFeedDataRequest{Head: header, Body: body}

	data, _ := json.Marshal(MarketFeedRequest)

	payload := strings.NewReader(string(data))

	bodyString := HTTPClient(method, url, payload)
	json.NewEncoder(w).Encode(bodyString)

	var MarketFeedRes MarketFeedResponse

	// json to golang struct
	json.Unmarshal([]byte(bodyString), &MarketFeedRes)
	fmt.Printf("TimeStamp: %s, StatusDescription: %s", MarketFeedRes.Body.TimeStamp, MarketFeedRes.Head.StatusDescription)

}
